x-pack/plugins/ml/common/constants/data_frame_analytics.ts

```diff
@@ -10,3 +10,12 @@ export const ANALYSIS_CONFIG_TYPE = {
   CLASSIFICATION: 'classification',
 } as const;
 export const DEFAULT_RESULTS_FIELD = 'ml';
+
+export const JOB_MAP_NODE_TYPES = {
+  ANALYTICS: 'analytics',
+  TRANSFORM: 'transform',
+  INDEX: 'index',
+  INFERENCE_MODEL: 'inferenceModel',
```

Comment:
'Trained model' is the term we are generally using now, so should probably rename these to be consistent.